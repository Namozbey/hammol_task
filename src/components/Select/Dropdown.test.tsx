import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import { Dropdown } from ".";

const options = [
  { label: "All", value: "all", sortOrder: "asc" },
  { label: "Docs", value: "doc", sortOrder: "asc" },
];

describe("Testing Dropdown", () => {
  it("Snapshots", () => {
    const { asFragment } = render(<Dropdown value="doc" options={options} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("Fire change event", () => {
    const onChange = (val: string, elm: DropdownOption) => {
      expect(val).toBe("doc");
      expect(elm).toEqual(options[1]);
      console.log("worked");
    };

    const { getByTestId } = render(
      <Dropdown data-testid="dropdown" options={options} onChange={onChange} />
    );
    const element = getByTestId("dropdown");
    fireEvent.click(element);
    fireEvent.click(screen.getByText("Docs"));
  });
});
