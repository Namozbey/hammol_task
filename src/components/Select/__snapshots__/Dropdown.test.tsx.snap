// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Dropdown Snapshots 1`] = `
<DocumentFragment>
  <div
    class="sc-jSMfEi brSdaV"
  >
    <div
      class="sc-gsnTZi evzPxy"
    >
      Docs
    </div>
    <span
      class="sc-dkzDqf ewRrEr material-icons"
    >
      arrow_drop_down
    </span>
    <div
      class="sc-hKMtZM bATtUK"
    >
      <label
        class="sc-eCYdqJ gYrPwk"
        font-size="20px"
      >
        <p>
          All
        </p>
      </label>
      <label
        class="sc-eCYdqJ laJkZi"
        font-size="20px"
      >
        <p>
          Docs
        </p>
      </label>
    </div>
  </div>
</DocumentFragment>
`;
